src/integer/poly_over_z/from.rs

```diff
@@ -83,6 +86,79 @@ impl FromStr for PolyOverZ {
     }
 }
 
+impl PolyOverZ {
+    /// Create a [`PolyOverZ`] from a [`PolyOverZq`].
+    ///
+    /// Parameters:
+    /// - `poly`: the polynomial from which the coefficients are copied
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer::PolyOverZ;
+    /// use qfall_math::integer_mod_q::PolyOverZq;
+    /// use std::str::FromStr;
+    ///
+    /// let poly = PolyOverZq::from_str("4  0 1 102 3 mod 101").unwrap();
+    ///
+    /// let poly_z = PolyOverZ::from_poly_over_zq(&poly);
+    ///
+    /// # let cmp_poly = PolyOverZ::from_str("4  0 1 1 3").unwrap();
+    /// # assert_eq!(cmp_poly, poly_z);
+    /// ```
+    pub fn from_poly_over_zq(poly: &PolyOverZq) -> Self {
+        let mut out = Self::default();
+        unsafe {
+            fmpz_mod_poly_get_fmpz_poly(
+                &mut out.poly,
+                &poly.poly,
+                poly.modulus.get_fmpz_mod_ctx_struct(),
+            )
+        };
+        out
+    }
+
+    /// Create a [`PolyOverZ`] from a [`PolynomialRingZq`].
+    ///
+    /// Parameters:
+    /// - `poly_ring`: the polynomial from which the coefficients are copied
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer::PolyOverZ;
+    /// use qfall_math::integer_mod_q::{PolynomialRingZq, ModulusPolynomialRingZq};
+    /// use std::str::FromStr;
+    ///
+    /// let modulus = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();
+    /// let poly = PolyOverZ::from_str("4  -1 0 1 1").unwrap();
+    /// let poly_ring = PolynomialRingZq::from((&poly, &modulus));
+    ///
+    /// let poly_z = PolyOverZ::from_polynomial_ring_zq(&poly_ring);
+    ///
+    /// # let cmp_poly = PolyOverZ::from_str("3  15 0 1").unwrap();
+    /// # assert_eq!(cmp_poly, poly_z);
+    /// ```
+    pub fn from_polynomial_ring_zq(poly_ring: &PolynomialRingZq) -> Self {
+        poly_ring.poly.clone()
+    }
```

Comment:
I don't think that we should provide this functionality and instead a getter or make the fields of the struct public.

Reply:
I used getter